### PR TITLE
teleport.service: Secure SSH access via Teleport

### PR DESCRIPTION
**Pull request by tuladhar**
Towards https://github.com/giantswarm/roadmap/issues/2712

This PR:

- Add teleport support to CAPVCD cluster

### Testing

Description on how {APP-NAME} can be tested.

- [x] fresh install works
- [x] upgrade from previous version works

#### Other testing

Description of features to additionally test for {APP-NAME} installations.

- [ ] check reconciliation of existing resources after upgrading
- [ ] X still works after upgrade
- [ ] Y is installed correctly

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.
- [x] Update `/examples` if required.

### CODE REVIEW

File: helm/cluster-cloud-director/values.yaml [hunk 1]
```diff
@@ -61,6 +61,10 @@ internal:
     name: giantswarm/pause
     registry: quay.io
     tag: "3.9"
+  teleport:
+    enabled: false
+    proxyAddr: test.teleport.giantswarm.io:443
+    version: 13.1.5
```
Comment (kopiczko) [inline helm/cluster-cloud-director/values.yaml:67]: This version is rather dated already.

File: helm/cluster-cloud-director/files/opt/teleport-installer.sh
Comment (kopiczko) [inline helm/cluster-cloud-director/files/opt/teleport-installer.sh:1]: Shouldn't it be baked into the image like for AWS?

Comment (vxav) [reply helm/cluster-cloud-director/files/opt/teleport-installer.sh:1]: CAPVCD is still a bit of a special snowflake as you know.
but flatcar support will arrive soon and should give us freedom over the image

**Comment by erkanerol** [comment]
This PR broke something in our test MC. Re-running tests to see the reason. See https://gigantic.slack.com/archives/C01BYMF6RN0/p1705046980987149

/run cluster-test-suites

**Comment by vxav** [comment]
Note that the WC testing is currently disabled in CAPVCD because Gerbil is unhealthy
https://gigantic.slack.com/archives/C04KA9DBMB2/p1706126086707949?thread_ts=1706125958.602149&cid=C04KA9DBMB2

**Comment by tuladhar** [comment]
## Teleport SSH works

Screenshot 2024-04-04 at 11 31 32 PM

**Review by vxav — APPROVED** [review]

**Pull request merged**